src/controllers/ControllerItems.js

```diff
@@ -0,0 +1,191 @@
+    });
+  } else {
+    item.properties._tofix = [{
+      action: data.action,
+      user: data.user,
+      time: now,
+      editor: data.editor
+    }];
+  }
+  item.properties._time = now + config.lockPeriod;
+  client.update({
+    index: 'tofix',
+    type: idtask,
+    id: item.properties._key,
+    body: {
+      doc: {
+        properties: item.properties
+      }
+    }
+  }, done);
+};
+
+module.exports.getItem = function(request, reply) {
+  var now = Math.round((new Date()).getTime() / 1000);
+  var idtask = request.params.idtask;
+  client.search({
+    index: 'tofix',
+    type: idtask,
+    body: {
+      size: 1,
+      sort: {
+        'properties._time': {
+          order: 'asc'
+        }
+      },
+      query: {
+        match_all: {}
+      }
+    }
+  }, function(err, resp) {
+    if (err) return reply(boom.badRequest(err));
+    var item = resp.hits.hits[0]._source;
+    reply(item);
+    //we know all new request is for edition
+    request.payload.action = 'edit';
+    request.payload.key = item.properties._key;
+    updateItemEdit(request, reply, item, now, function(err, resp) {
+      if (err) console.log(err);
+      updateStatsInTask(request, reply);
+      updateActivity(request, reply, now);
+    });
+  });
+};
+
+module.exports.getItemById = function(request, reply) {
+  var key = request.params.key;
+  var idtask = request.params.idtask;
+  client.get({
+    index: 'tofix',
+    type: idtask,
+    id: key
+  }, function(err, resp) {
+    if (err) return reply(boom.badRequest(err));
+    reply(resp._source);
+  });
+};
+
+module.exports.updateItem = function(request, reply) {
+  var idtask = request.params.idtask;
+  var data = request.payload;
+  var key = data.key;
+  var now = Math.round((new Date()).getTime() / 1000);
+  client.get({
+    index: 'tofix',
+    type: idtask,
+    id: key
+  }, function(err, resp) {
+    if (err) return reply(boom.badRequest(err));
+    var item = resp._source;
+    item.properties._tofix.push({
+      action: data.action,
+      user: data.user,
+      time: now,
+      editor: data.editor
+    });
+    var maxNum = config.maxNum;
+    if (data.action === 'skip') {
+      maxNum = now + config.lockPeriod;
+    }
+    item.properties._time = maxNum;
+    client.update({
+      index: 'tofix',
+      type: idtask,
+      id: key,
+      body: {
+        doc: {
+          properties: item.properties
+        }
+      }
+    }, function(err, resp) {
+      if (err) return reply(boom.badRequest(err));
+      reply({
+        index: resp._index,
+        type: resp._type,
+        key: resp._id,
+        status: 'updated'
+      });
+      updateStatsInTask(request, reply);
+      updateActivity(request, reply, now);
+    });
+  });
+};
+
+module.exports.getAllItems = function(request, reply) {
+  var idtask = request.params.idtask;
+  client.search({
+    index: 'tofix',
+    type: idtask,
+    body: {
+      size: 1000, //get item using pagination - need to fix here
```

Comment:
We could probably default this to a lower number, like 50 or 100, to keep the frontend responsive. RE pagination: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-from-size.html